services/client/src/api/pools/useAddHypePool.tsx

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+import { AddHypePool, HypePool } from '../../models';
+import { UnpackNestedValue } from 'react-hook-form';
+import useWallet from '../../hooks/useWallet';
+import { API } from './types';
+
+const postNewPool = async (pool: HypePool) => {
+  const url = `${API}/pools`;
+  const { data } = await axios.post(url, pool);
+  return data as HypePool;
+};
+
+export const useAddHypePool = () => {
```

Comment:
I think as it is, this is a "false hook". We could refactor it into a proper hook and then use it as `use` in thw following setup:


```suggestion
const api = useApi(<endpoint>)

export const usePoolApi(endpoint: string) {
 const [error, setError] = useEffect({});
 const addHypePool() {
  ...
  }
  
  const fetchHypePool() {
   ...
  }
}

```

Reply:
Actually each API call like in this case should be a separate hook. If we changed them like this example, they will be just normal methods, react-query won't call them on component mount. So I think we should leave them like this.

Reply:
OKay, that clarifies.